src/impi/core.clj

```diff
@@ -0,0 +1,12 @@
+(ns impi.core)
+
+(defmacro add-props! [& {:as coll}]
```

Comment:
Why are you using a macro here?

Rather than creating many multimethods, we could have one that looks up the value in a map:

```clojure
(def ^:private text-properties
  {:pixi.text.style/align "align"})

(doseq [attr (keys text-properties)]
  (derive attr :pixi.text.style/property))

(defmethod update-prop! :pixi.text.style/property [object index attr value]
  (aset object (text-properties attr) value))
```

That should be just as performant, as multimethods look up their dispatch keys via a map anyway.